common/safe_future_client.h

```diff
@@ -183,16 +184,16 @@ class SafeFutureClient {
       return;
     }
 
-    try {
-      typename RequestTraits<T>::response_type response;
-      RequestTraits<T>::recv_func(response, *recvState);
-      p->setValue(std::move(response));
-    } catch (typename RequestTraits<T>::exception_type& e) {
-      p->setException(
-        folly::make_exception_wrapper<
-          typename RequestTraits<T>::exception_type>(std::move(e)));
-    } catch (std::exception& e) {
-      p->setException(folly::exception_wrapper(std::move(e)));
+    auto ex = folly::try_and_catch<std::exception,
+                                   apache::thrift::TApplicationException,
+                                   typename RequestTraits<T>::exception_type>([&p, &recvState]() {
```

Comment:
nit: can we capture by value instead of capture by reference? which can reduce one level of indirect access.

Reply:
sure. just found p is a pointer..